pandas/core/internals/construction.py

```diff
@@ -400,7 +400,15 @@ def _prep_ndarray(values, copy: bool = True) -> np.ndarray:
             return arr[..., np.newaxis]
 
         def convert(v):
-            return maybe_convert_platform(v)
+            if not is_list_like(v) or isinstance(v, ABCDataFrame):
```

Comment:
this seems like a pretty general function (just commenting nothing to do now)

Reply:
yah im hoping we'll end up with just one sanitize_array-like function by the end of this process.  the hard part at this point is deciding what we want to do about #40114, #40110, and #24435